### PR TITLE
define new keyboard "suihankey"

### PR DESCRIPTION
**Pull request by kakunpc**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+#define KC_CMD_SP    LT(COMMAND,KC_SPC)
+#define KC_CMD_ET    LT(COMMAND,KC_ENTER)
+#define KC_NUM_ALT   LT(NUMBER,KC_LALT)
+#define KC_NUM_BS    LT(NUMBER,KC_BSPC)
+#define KC_SET_CTRL  LT(SETTING,KC_LCTRL)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [BASE] = LAYOUT( /* Base */
+          KC_Q,  KC_W,  KC_E, KC_R, KC_T,                 KC_Y,  KC_U,  KC_I, KC_O, KC_P,\
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,                 KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,\
+          KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,                 KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,\
+          KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,                 KC_NO,  KC_NO,  KC_NO, KC_NO, KC_NO,\
+                        KC_NO,  KC_NO,  KC_NO,                 KC_NO,  KC_NO,  KC_NO   \
+  ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QMKBEST:
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]: Looks like you can remove these two keycodes as they aren't being used in the keymap.

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    case SETTING:
+      oled_write_P(PSTR("SETTING\n"), false);
+      break;
+    default:
+      // Or use the write_ln shortcut over adding '\n' to the end of your string
+      oled_write_ln_P(PSTR("Undefined"), false);
+  }
+
+  // Host Keyboard LED Status
+  uint8_t led_usb_state = host_keyboard_leds();
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  oled_write_P(led_usb_state & (1<<USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_HOST_LED_ON(USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
```

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  oled_write_P(led_usb_state & (1<<USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_HOST_LED_ON(USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
```

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  oled_write_P(led_usb_state & (1<<USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(IS_HOST_LED_ON(USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```

**Comment by kakunpc** [comment]
I fixed it.

File: keyboards/suihankey/keymaps/default/config.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// place overrides here
+#if RGBLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/suihankey/keymaps/default/config.h]: While, normally, this is absolutely fine here.... the one issue is that when using the configurator, it won't be used. 

Ideally, this should actually be in the keyboard's `config.h` file, and not here.

File: keyboards/suihankey/config.h [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+   #define RGBLED_NUM 18
+   #define RGBLIGHT_HUE_STEP 8
+   #define RGBLIGHT_SAT_STEP 8
+   #define RGBLIGHT_VAL_STEP 8
+   #define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+   #define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+// /*== all animations enable ==*/
+//   #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/suihankey/config.h]:
```suggestion
    #define RGBLIGHT_ANIMATIONS
```

File: keyboards/suihankey/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# suihankey
+
+![suihankey](imgur.com image replace me!)
```
Comment (drashna) [inline keyboards/suihankey/readme.md]: Do you have an image you can use to replace this?

**Comment by kakunpc** [comment]
I fixed it.

OLED is portrait mode.

File: keyboards/suihankey/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# suihankey
+
+![suihankey](https://i.gyazo.com/f798c5967f2ac457dd520ab8ff83b6ac.jpg)
+
+A short description of the keyboard/project
```
Comment (mechmerlin) [inline keyboards/suihankey/readme.md]: Can you provide a short description of the keyboard/project here?

File: keyboards/suihankey/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+Keyboard Maintainer: [kakunpc](https://github.com/kakunpc)  
+Hardware Supported: The PCBs, controllers supported
```
Comment (mechmerlin) [inline keyboards/suihankey/readme.md]: Can you fill this out too?

File: keyboards/suihankey/readme.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# suihankey
+
+![suihankey](https://i.gyazo.com/f798c5967f2ac457dd520ab8ff83b6ac.jpg)
+
+A short description of the keyboard/project
+
+Keyboard Maintainer: [kakunpc](https://github.com/kakunpc)  
+Hardware Supported: The PCBs, controllers supported  
+Hardware Availability: links to where you can find this hardware
```
Comment (mechmerlin) [inline keyboards/suihankey/readme.md]: Can you fill this out as well?

**Comment by drashna** [comment]
Awesome! 

As for the OLED, the `OLED_DRIVER` feature actually supports rotating the screen.  The caveat is that you only get about 5 columns of characters, per each line.   

But for simplicity, you can use `oled_write_p_ln`, and that will "end the line" for you. 

Also, if you take a look at my corne (crkbd) keymap, you can see a lot more of what you can do there. 

IMG_20190624_100429

But this is entirely optional.  You don't have to do any of this, if you don't want.  But it is useful, as you can actually see it.

**Comment by kakunpc** [comment]
README.md updated.
We've also added info.json.

@drashna 
Thanks for the suggestion.
OLED is implemented in portrait, but there are no plans to rotate it for now.

File: keyboards/suihankey/readme.md [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+![suihankey](https://i.gyazo.com/f798c5967f2ac457dd520ab8ff83b6ac.jpg)
+
+Compact with only 36 keys is a concept keyboard.  
+Supports OLED and RGBLED (optional)
+
+
+
+Keyboard Maintainer: [kakunpc](https://github.com/kakunpc)  
+Hardware Supported: suihankeyboard_alpha, promicro
```
Comment (fauxpark) [inline keyboards/suihankey/readme.md]:
```suggestion
Hardware Supported: suihankeyboard_alpha, promicro  
```

The two spaces at the end of this line are required for proper rendering.

File: keyboards/suihankey/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+  ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+void matrix_init_user(void) {
+#if RGBLIGHT_ENABLE
+    rgblight_enable_noeeprom();
```
Comment (fauxpark) [inline keyboards/suihankey/keymaps/default/keymap.c]: I'm fairly certain this isn't necessary, as rgblight is not enabled by default, and there are no RGB keycodes assigned in this keymap.

**Comment by kakunpc** [comment]
If you don't mind, marge, please!

**Comment by drashna** [comment]
Thanks!

**Pull request merged**